### PR TITLE
[RFC] Fix lint errors

### PR DESCRIPTION
**Pull request by steveno**
Remove formatting errors from arabic.c and cursor_shape.c and remove clint ignore.

NOTE: This is more than just white space since I had add some brackets and change FALSE to false.

### CODE REVIEW

File: src/nvim/cursor_shape.c [hunk 1]
```diff
@@ -85,18 +85,18 @@ char_u *parse_shape_opt(int what)
...
+              if (STRNICMP(modep, shape_table[idx].name, len) == 0)
                 break;
-            if (idx == SHAPE_IDX_COUNT
-                || (shape_table[idx].used_for & what) == 0)
+          
+	    if (idx == SHAPE_IDX_COUNT || (shape_table[idx].used_for & what) == 0)
               return (char_u *)N_("E546: Illegal mode");
-            if (len == 2 && modep[0] == 'v' && modep[1] == 'e')
-              found_ve = TRUE;
+	    if ((len = 2 && modep[0] == 'v' && modep[1] == 'e'))
```
Comment (justinmk) [inline src/nvim/cursor_shape.c]: Bug

Comment (justinmk) [inline src/nvim/cursor_shape.c]: Also why would lint require the extra parens?

Comment (steveno) [inline src/nvim/cursor_shape.c]: The extra parens weren't from the lint; they were from GCC actually. The compiler told me that the statement was ambiguous.

Comment (steveno) [inline src/nvim/cursor_shape.c]: @justinmk What's the bug exactly?

**Comment by steveno** [comment]
I have fixed the noted issues.

**Comment by steveno** [comment]
I don't understand why it "FAILED". When I try to look through the reports all I see are Java errors that appear completely unrelated to neovim...

File: src/nvim/cursor_shape.c [hunk 2]
```diff
@@ -85,18 +85,18 @@ char_u *parse_shape_opt(int what)
...
-                  == 0)
+              if (STRNICMP(modep, shape_table[idx].name, len) == 0)
                 break;
             if (idx == SHAPE_IDX_COUNT
-                || (shape_table[idx].used_for & what) == 0)
+                    || (shape_table[idx].used_for & what) == 0)
               return (char_u *)N_("E546: Illegal mode");
-            if (len == 2 && modep[0] == 'v' && modep[1] == 'e')
-              found_ve = TRUE;
+            if ((len == 2 && modep[0] == 'v' && modep[1] == 'e'))
```
Comment (justinmk) [inline src/nvim/cursor_shape.c]: Please remove the extra parens.

**Comment by steveno** [comment]
Pushed again. Sorry about the parens.

**Comment by justinmk** [comment]
Merged before #3185 to see the result on bot-ci.

**Pull request closed without merging**